src/main/java/gyro/google/storage/BucketEncryption.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.storage;
+
+import com.google.api.services.storage.model.Bucket.Encryption;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Updatable;
+import gyro.google.Copyable;
+
+/**
+ * Subresource for setting the Bucket.Encryption configuration to a Bucket.
+ */
+public class BucketEncryption extends Diffable implements Copyable<Encryption> {
+
+    private String defaultKmsKeyName;
+
+    /**
+     * Default KMS key used to encrypt objects inserted into the bucket when no encryption method is specified.
+     */
+    @Updatable
```

Comment:
Must have an explicit primary key, otherwise changes to this updatable field triggers a create/delete flow

Reply:
I am going to remove BucketEncryption. This needs to be a reference the a Key resource that has yet to be built.